### PR TITLE
Sixth day

### PR DESCRIPTION
**Pull request by hrodberaht**

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -3,7 +3,5 @@ import ReactDOM from 'react-dom';
 import App from './App';
 
 it('renders without crashing', () => {
```
Comment (ar-mac) [inline src/App.test.js:5]: This test case is obsolete now and can be removed

File: src/actions/index.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import * as actions from "./index";
+
+describe("cards actions", () => {
+  it("addCard should create addCard action", () => {
+    expect(actions.addCard("John", "test@test.pl")).toEqual({
```
Comment (ar-mac) [inline src/actions/index.test.js:5]: You can also use snapshots here.
`expect(actions.addCard("John", "test@test.pl")).toMatchSnapshot()`
it's especially useful when the action returns a lot of data. In such small examples, it's usability is much smaller.
Also, when using snapshot tests remember to always check if they are correct when you generate them first, as the action could return some bad data and the snapshot would happily save that result assuming it's what you needed.

File: src/reducers/cards.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    }
+  ];
+
+  it("should have initialstate", () => {
+    expect(cards(undefined, {})).toEqual(initialState);
+  });
+
+  it("should add card to cards", () => {
+    expect(
+      cards(initialState, {
```
Comment (ar-mac) [inline src/reducers/cards.test.js:23]: Here you could use action creator to generate the action you have here. That way you could confirm action creator validity. (Then you would not need the separate specs for action creators)

File: src/reducers/cards.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        type: "ADD_CARD",
+        id: 3,
+        name: "Joh",
+        email: "test@test.pl"
+      })
+    ).toEqual([
```
Comment (ar-mac) [inline src/reducers/cards.test.js:29]: You could check only the relevant part using proper selectors. In this case https://jestjs.io/docs/en/expect#expectarraycontainingarray

File: src/reducers/cards.test.js (hunk 3 above)
Comment (hrodberaht) [reply src/reducers/cards.test.js:23]: you mean testing in one file redurcer and action?

File: src/reducers/cards.test.js (hunk 4 above)
Comment (hrodberaht) [reply src/reducers/cards.test.js:29]: ok

File: src/actions/index.test.js (hunk 2 above)
Comment (hrodberaht) [reply src/actions/index.test.js:5]: If i use snapshot where id is changed, the snapshot, don't match,

Comment (ar-mac) [reply src/actions/index.test.js:5]: But now we are expecting id to be 3, so it's not much different than the snapshot.
If the ID would be generated from date or from some hashing mechanism we could work around it as well by making it always return the same value.
Another solution is to use matcher 
```
.toMatchObject({
 type: "ADD_CARD",
 name: "John",
 email: "test@test.pl",
})
```
and not specify value of ID

File: src/reducers/cards.test.js (hunk 3 above)
Comment (ar-mac) [reply src/reducers/cards.test.js:23]: I mean, by using action creators to generate action object in this spec. We confirm that action creator works properly.
As this is reducer spec, we also confirm reducer logic. So in this one test we can confirm that 2 things work together properly. That way it's not (strictly speaking) unit test, but the borders between types sometimes are blurred and we do not need to conform to one or another.

**Pull request merged**